pac4j-core/src/main/java/org/pac4j/core/profile/InternalAttributeHandler.java

```diff
@@ -0,0 +1,128 @@
+package org.pac4j.core.profile;
+
+import org.pac4j.core.profile.converter.Converters;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.core.util.JavaSerializationHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Internally handles attributes (set / get).
+ *
+ * @author Jerome Leleu
+ * @since 2.0.0
+ */
+public class InternalAttributeHandler implements Serializable {
+
+    private static final long serialVersionUID = 3391226300882369266L;
+
+    protected transient final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public transient final static String PREFIX = "{#";
+    public transient final static String PREFIX_BOOLEAN = PREFIX + "bool}";
+    public transient final static String PREFIX_INT = PREFIX + "int}";
+    public transient final static String PREFIX_LONG = PREFIX + "long}";
+    public transient final static String PREFIX_DATE = PREFIX + "date}";
+    public transient final static String PREFIX_URI = PREFIX + "uri}";
+    public transient final static String PREFIX_SB64 = PREFIX + "sb64}";
+
+    private JavaSerializationHelper serializationHelper = new JavaSerializationHelper();
+
+    private boolean stringify = false;
+
+    /**
+     * Before saving the attribute into the attributes map.
+     *
+     * @param value the original value
+     * @return the prepared value
+     */
+    public Object prepare(final Object value) {
+        if (value == null || value instanceof String || !stringify) {
+            return value;
+        } else {
+            if (value instanceof Boolean) {
+                return PREFIX_BOOLEAN + value;
+            } else if (value instanceof Integer) {
+                return PREFIX_INT + value;
+            } else if (value instanceof Long) {
+                return PREFIX_LONG + value;
+            } else if (value instanceof Date) {
+                return PREFIX_DATE + newSdf().format((Date) value);
+            } else if (value instanceof URI) {
+                return PREFIX_URI + value.toString();
+            } else {
+                return PREFIX_SB64 + serializationHelper.serializeToBase64((Serializable) value);
```

Comment:
OK. This probably is fine, but I would have preferred perhaps overloaded method so you could be a lot more explicit about the type rather than a generic object type.